### PR TITLE
Feature#12Home-view-frontend

### LINKED ISSUE #12: Home view frontend
# REQUIREMENTS
Create the view 'Home', this will the main view that the users, once login they will see. This view contain some videos about different topics to take.

# DESIGN
The view will looks as follows: 
Home - Mau

NOTE: All the space blanks will be replaced with real content.

### PR DESCRIPTION
**Pull request by MauricioNC**
It was created the Home view.
The videos for the classes was embeded, wich means that we don't have to
store them in our database.

# Description

 - It was created the home view, this view will be the main view that the users will see once login.
 - I created a template for the navbar, this template is linked to the home view, and will be linked the others views.

  ## Screenshots 
image
image

### CODE REVIEW

File: EasyLearn/app/assets/stylesheets/css/cards_classes.css [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  padding: 20px 60px;
+}
+
+.card-deck {
+  margin: 100px;
+}
+
+.card > iframe {
+  width: 100%;
+  height: 240px;
```
Comment (JuanequeX) [inline EasyLearn/app/assets/stylesheets/css/cards_classes.css]: Sort this block

File: EasyLearn/app/assets/stylesheets/css/cards_classes.css [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  border: 0px solid #FFF;
+  box-shadow: 4px 4px 15px 1px rgb(31, 31, 31);
+  background-color: #282828;
+}
+
+.card-body {
+  display: flex;
+  justify-content: space-between;
+  align-items: flex-start;
+  flex-direction: column;
```
Comment (JuanequeX) [inline EasyLearn/app/assets/stylesheets/css/cards_classes.css]: sort this block

File: EasyLearn/app/assets/stylesheets/css/cards_classes.css [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  color: #e0e0e0
+}
+
+.card-body .btn { background-color: #F78528; }
+
+.card .card-footer small {
+  display: flex;
+  justify-content: flex-start;
+  align-items: center;
+  flex-direction: row;
```
Comment (JuanequeX) [inline EasyLearn/app/assets/stylesheets/css/cards_classes.css]: Sort this block

File: EasyLearn/app/assets/stylesheets/home.css [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  background-color: #282828;
+  border-bottom: 1px solid #F78528;
+}
+
+.navbar .collapse form .form-control{
+  background-color: #212129;
+  border: none;
+  padding: 4px 15px;
+  color: rgb(163, 163, 163);
+  width: 330px;
```
Comment (JuanequeX) [inline EasyLearn/app/assets/stylesheets/home.css]: Sort this block

File: EasyLearn/app/assets/stylesheets/home.css [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+/* BANNER STYLES */
+
+.banner {
+  margin-bottom: 50px;
+  padding: 30px;
+  background-color: #2d2d33;
+  box-shadow: 6px 0px 10px 3px rgba(31, 31, 31, 0.473);
+}
```
Comment (JuanequeX) [inline EasyLearn/app/assets/stylesheets/home.css]: Sort this block

File: EasyLearn/app/assets/stylesheets/home.css [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+.banner .row {
+  display: flex;
+  justify-content: center;
+  align-content: center;
+  align-items: center;
+  flex-direction: row;
+  height: 100%;
```
Comment (JuanequeX) [inline EasyLearn/app/assets/stylesheets/home.css]: Sort this block

File: EasyLearn/app/assets/stylesheets/home.css [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+.banner .banner-text h1{
+  font-size: 3rem;
+}
+
+@supports(object-fit: cover){
+  .banner-img img{
+    width: 70%;
+    object-fit: contain;
+    object-position: center center;
```
Comment (JuanequeX) [inline EasyLearn/app/assets/stylesheets/home.css]: Sort this block

File: EasyLearn/app/assets/stylesheets/home.css [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+  }
+}
```
Comment (JuanequeX) [inline EasyLearn/app/assets/stylesheets/home.css]: Add a blank line here

File: EasyLearn/app/views/shared/_menu.html.erb [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+            <a class="dropdown-item" href="#">Settings</a>
+            <div class="dropdown-divider"></div>
+            <a class="dropdown-item" href="#">Sign out</a>
+          </div>
+        
+        </li>
+      </ul>
+
+    </div>
+  </nav>
```
Comment (JuanequeX) [inline EasyLearn/app/views/shared/_menu.html.erb]: Add a blank line here

File: EasyLearn/app/views/shared/home/_cards_classes.html.erb [hunk 10]
```diff
@@ -0,0 +1,68 @@
+<div class="cards-container">
+  <div class="card-deck">
+    
+    <div class="card">
+      <iframe src="https://www.youtube.com/embed/MfeOZooNamI" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+      <div class="card-body">
+        <h5 class="card-title">Implemented AJAX with PHP</h5>
+        <p class="card-text">Learn how to share information from AJAX to PHP and viceversa.</p>
+        <%= link_to "Take calss", "#", class: "btn" %>
```
Comment (JuanequeX) [inline EasyLearn/app/views/shared/home/_cards_classes.html.erb]: It should be like that `<%= link_to "Take class", "#", class: "btn" %>`, no?

File: EasyLearn/app/views/shared/home/_cards_classes.html.erb [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+    </div>
+    
+    <div class="card">
+      <iframe src="https://www.youtube.com/embed/GfCUnf2y4lY" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+      <div class="card-body">
+        <h5 class="card-title">Contigous allocation - Calloc</h5>
+        <p class="card-text">
+          Calloc, is another function that we can use for dynamic memory management, this works in a similar way to malloc.
+        </p>
+        <%= link_to "Take calss", "#", class: "btn" %>
```
Comment (JuanequeX) [inline EasyLearn/app/views/shared/home/_cards_classes.html.erb]: Same here

File: EasyLearn/app/views/shared/home/_cards_classes.html.erb [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+        </small>
+      </div>
+    </div>
+    
+    <div class="card">
+      <iframe src="https://www.youtube.com/embed/fmyvWz5TUWg" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+      <div class="card-body">
+        <h5 class="card-title">Rails - full course</h5>
+        <p class="card-text">Learn Rails from 0 in 4 hours.</p>
+        <%= link_to "Take calss", "#", class: "btn" %>
```
Comment (JuanequeX) [inline EasyLearn/app/views/shared/home/_cards_classes.html.erb]: and here

File: EasyLearn/app/views/shared/home/_cards_classes.html.erb [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+        <small class="text-muted">
+          <span class="material-icons icon-profile">account_circle</span>
+          <span class="owner-name">Alan</span>
+        </small>
+      </div>
+    </div>
+
+    
+  </div>
+</div>
```
Comment (JuanequeX) [inline EasyLearn/app/views/shared/home/_cards_classes.html.erb]: Add a blank line here

File: EasyLearn/app/views/shared/home/_cards_classes.html.erb [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+        </small>
+      </div>
+    </div>
+   
+    <div class="card">
+      <iframe src="https://www.youtube.com/embed/RqQ1d1qEWlE" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+      <div class="card-body">
+        <h5 class="card-title">Javascript</h5>
+        <p class="card-text">Learn JavaScript from 0.</p>
+        <%= link_to "Take calss", "#", class: "btn" %>
```
Comment (JuanequeX) [inline EasyLearn/app/views/shared/home/_cards_classes.html.erb]: `Class`

**Review by JuanequeX — CHANGES_REQUESTED** [review]

File: EasyLearn/app/helpers/home_helper.rb [hunk 15]
```diff
@@ -0,0 +1,2 @@
+module HomeHelper
+end
```
Comment (YilKanda) [inline EasyLearn/app/helpers/home_helper.rb]: If you don't use it, remove it.

File: EasyLearn/test/controllers/home_controller_test.rb [hunk 16]
```diff
@@ -0,0 +1,8 @@
+require "test_helper"
+
+class HomeControllerTest < ActionDispatch::IntegrationTest
+  test "should get index" do
+    get home_index_url
```
Comment (YilKanda) [inline EasyLearn/test/controllers/home_controller_test.rb]:
```suggestion
    get home_url
```

File: EasyLearn/app/views/shared/home/_cards_classes.html.erb [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+      </div>
+      <div class="card-footer">
+        <small class="text-muted">
+          <span class="material-icons icon-profile">account_circle</span>
+          <span class="owner-name">Alan</span>
+        </small>
+      </div>
+    </div>
+
+
```
Comment (YilKanda) [inline EasyLearn/app/views/shared/home/_cards_classes.html.erb]: Remove unnecessary spaces

File: EasyLearn/app/views/shared/home/_cards_classes.html.erb [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+        <%= link_to "Take calss", "#", class: "btn" %>
+      </div>
+      <div class="card-footer">
+        <small class="text-muted">
+          <span class="material-icons icon-profile">account_circle</span>
+          <span class="owner-name">Alan</span>
+        </small>
+      </div>
+    </div>
+
```
Comment (YilKanda) [inline EasyLearn/app/views/shared/home/_cards_classes.html.erb]: When there are repeating elements that have the same structure and only the text changes, you can make use of the [internationalization](https://guides.rubyonrails.org/i18n.html) property to simplify your code.

Comment (MauricioNC) [reply EasyLearn/app/views/shared/home/_cards_classes.html.erb]: I read some about this property, but I think that is something more advanced for us, but thanks for the advice, we work on it to try to solve this inconvenient.

File: EasyLearn/app/views/shared/home/_cards_classes.html.erb (hunk 17 above)
Comment (MauricioNC) [reply EasyLearn/app/views/shared/home/_cards_classes.html.erb]: Done.

File: EasyLearn/test/controllers/home_controller_test.rb (hunk 16 above)
Comment (MauricioNC) [reply EasyLearn/test/controllers/home_controller_test.rb]: Done.

File: EasyLearn/app/helpers/home_helper.rb (hunk 15 above)
Comment (MauricioNC) [reply EasyLearn/app/helpers/home_helper.rb]: Done.

File: EasyLearn/app/views/shared/home/_cards_classes.html.erb (hunk 14 above)
Comment (MauricioNC) [reply EasyLearn/app/views/shared/home/_cards_classes.html.erb]: Done.

File: EasyLearn/app/views/shared/home/_cards_classes.html.erb (hunk 13 above)
Comment (MauricioNC) [reply EasyLearn/app/views/shared/home/_cards_classes.html.erb]: Done.

File: EasyLearn/app/views/shared/home/_cards_classes.html.erb (hunk 12 above)
Comment (MauricioNC) [reply EasyLearn/app/views/shared/home/_cards_classes.html.erb]: Done.

File: EasyLearn/app/views/shared/home/_cards_classes.html.erb (hunk 11 above)
Comment (MauricioNC) [reply EasyLearn/app/views/shared/home/_cards_classes.html.erb]: Done.

File: EasyLearn/app/views/shared/home/_cards_classes.html.erb (hunk 10 above)
Comment (MauricioNC) [reply EasyLearn/app/views/shared/home/_cards_classes.html.erb]: Done.

File: EasyLearn/app/assets/stylesheets/home.css (hunk 5 above)
Comment (MauricioNC) [reply EasyLearn/app/assets/stylesheets/home.css]: Done.

File: EasyLearn/app/assets/stylesheets/home.css (hunk 4 above)
Comment (MauricioNC) [reply EasyLearn/app/assets/stylesheets/home.css]: Done.

File: EasyLearn/app/assets/stylesheets/css/cards_classes.css (hunk 3 above)
Comment (MauricioNC) [reply EasyLearn/app/assets/stylesheets/css/cards_classes.css]: Done.

File: EasyLearn/app/assets/stylesheets/css/cards_classes.css (hunk 2 above)
Comment (MauricioNC) [reply EasyLearn/app/assets/stylesheets/css/cards_classes.css]: Done.

File: EasyLearn/app/assets/stylesheets/css/cards_classes.css (hunk 1 above)
Comment (MauricioNC) [reply EasyLearn/app/assets/stylesheets/css/cards_classes.css]: Done.

**Review by JuanequeX — APPROVED** [review]
LGTM

**Review by VictorElizalde — APPROVED** [review]
LGTM, just solve conflicts for merge.

**Review by YilKanda — APPROVED** [review]
LGTM!

**Pull request merged**